### PR TITLE
Classes 11-16

### PR DESCRIPTION
**Pull request by jimflores5**
Please do a careful review of classes 13 - 16 and a cursory review of classes 11 & 12.

### CODE REVIEW

File: Class-12-(HTML-Git-Terminal).md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+   1. Note tedious aspects as they appear (e.g. making EVERY h2 red), and note that we will learn how to streamline this process in the next class.
+1. Reinforce why using a VCS is a good idea. If you have a personal story that relates to this, feel free to share it!
+   1. If time, initialize a simple git repo for the web page you created. Show how edits (or mistakes) can be undone with git.
+1. Studio intro. Task uses browser developer tools.
+
+## Studio (TA Notes)
+
+1. Common student stumbling blocks:
+   1. Finding the dev tools for their particular browser version.
+   1. Using copyrighted images. This studio probably falls under fair use so long as the work does not get posted elsewhere. However, it would be a good idea to remind your students that just because something is found on the web, that does NOT mean it's free to use. If in doubt, DON'T.
```
Comment (gildedgardenia) [inline Class-12-(HTML-Git-Terminal).md]: I might just suggest using only personal images. The grey area is very big here!

File: Class-12-(HTML-Git-Terminal).md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+1. Encourage having fun! Ooooh and aaaaah as you check in with each of your students.
```
Comment (gildedgardenia) [inline Class-12-(HTML-Git-Terminal).md:36]: I would also remind them to make sure their students keep it light-hearted. The studio page has some good language around this.

File: Class-13-(CSS-Git).md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+## In the prep work this week (CSS, More Git), the students learned:
+
+1. CSS:
+   1. Syntax.
+   1. How to set font-size, color, font-family, background-color, text-align, etc.
+   1. Assign styles to an element, class, or ID.
+   1. Internal vs. inline vs. external locations.
+1. More Git:
+   1. Clone a remote repo, or create a remote repo from a local one,
```
Comment (gildedgardenia) [inline Class-13-(CSS-Git).md]: Lines 9 and 10 should end in periods!

File: Class-15-(HTTP-&-Forms).md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+1. Review the differences between GET and POST. Why should one method be used over the other?
+1. How to use event handlers to validate form data.
+1. Live coding:
+    1. Show how changing ``type = "text"`` to ``type = "email"`` adds basic validation to an HTML ``input``, even without an event handler.
+    1. Difference between ``radio``, ``select``, and ``checkbox``.
+    1. How to align multiple labels and inputs:
+        1. Horizontally,
+        1. Vertically,
+        1. Centered within a ``div``.
+1. If time, discuss why basic HTML forms (inputs, buttons, etc.) do not usually look like the ones we see on company websites.
```
Comment (gildedgardenia) [inline Class-15-(HTTP-&-Forms).md]: "If time allows, ..."

File: Class-15-(HTTP-&-Forms).md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+1. Studio intro.
+1. *Studio post-op. This activity will probably require a debrief at the end of the class*.
```
Comment (gildedgardenia) [inline Class-15-(HTTP-&-Forms).md]: "Studio follow-up"

File: Class-16-(Fetch-&-JSON).md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    1. Syntax,
+    1. How to use the dev tools to view what ``fetch`` does,
+    1. The ``response`` keyword,
+    1. How to use the response from an API to modify the HTML on a webpage.
+1. Studio intro.
+
+## Studio (TA Notes)
+
+1. Forking and cloning a GitHub repository should be old hat by now. If a student requires help, point them to a book example that shows how to fork and clone rather than walking them through the process again.
+1. Absolutely, positively, NO EXCUSES, complete this studio on your own before class.
```
Comment (gildedgardenia) [inline Class-16-(Fetch-&-JSON).md]: "NO EXCUSES! Complete ..."

File: Class-16-(Fetch-&-JSON).md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    1. There are lots of tiny details that students may miss or mistype that cause their code to break.
+    1. When a student asks for help, have them describe in detail what they have done, what was working before, and where they think their code might be misbehaving.
+1. Some students might get confused about the sample JSON given in the instructions. Remind them that they do NOT need to cut and paste this data. Instead, their program will request data from an API.
+1. Similarly, students might get confused about where to add the HTML for the astronaut data.
+1. As always, encourage your group to try the Bonus missions or play with the CSS file to try to change the appearance of the webpage.
```
Comment (gildedgardenia) [inline Class-16-(Fetch-&-JSON).md]: "the Bonus missions, or play with the CSS file ..."

**Review by gildedgardenia — CHANGES_REQUESTED** [review]
Overall, this looks awesome! Just a few small grammar issues!

File: Class-12-(HTML-Git-Terminal).md (hunk 1 above)
Comment (jimflores5) [reply Class-12-(HTML-Git-Terminal).md]: Rephrased.  Thanks.

File: Class-12-(HTML-Git-Terminal).md (hunk 2 above)
Comment (jimflores5) [reply Class-12-(HTML-Git-Terminal).md:36]: Done.  Thanks.

File: Class-13-(CSS-Git).md (hunk 3 above)
Comment (jimflores5) [reply Class-13-(CSS-Git).md]: Fixed.  Thanks.

File: Class-15-(HTTP-&-Forms).md (hunk 4 above)
Comment (jimflores5) [reply Class-15-(HTTP-&-Forms).md]: Fixed.  Thanks.

File: Class-15-(HTTP-&-Forms).md (hunk 5 above)
Comment (jimflores5) [reply Class-15-(HTTP-&-Forms).md]: Changed.  Thanks.

File: Class-16-(Fetch-&-JSON).md (hunk 6 above)
Comment (jimflores5) [reply Class-16-(Fetch-&-JSON).md]: I was wondering about the punctuation here.  Thanks for the assist!

"That that is is that that is not is not"

File: Class-16-(Fetch-&-JSON).md (hunk 7 above)
Comment (jimflores5) [reply Class-16-(Fetch-&-JSON).md]: Fixed.  Thanks.

**Review by gildedgardenia — APPROVED** [review]
Looks great!

**Pull request merged**